### PR TITLE
upgrade rosetta-sdk,iotex-core etc.

### PR DESCRIPTION
**Pull request by millken**

### CODE REVIEW

File: rosetta-cli-config/mainnet/iotex.json [hunk 1]
```diff
@@ -6,87 +6,36 @@
...
+ "memory_limit_disabled": false,
+ "construction": null,
  "data": {
   "active_reconciliation_concurrency": 16,
   "inactive_reconciliation_concurrency": 4,
   "inactive_reconciliation_frequency": 250,
   "log_blocks": false,
   "log_transactions": false,
-  "log_balance_changes": true,
+  "log_balance_changes": false,
```
Comment (dustinxie) [inline rosetta-cli-config/mainnet/iotex.json:25]: why change to false?

File: rosetta-cli-config/mainnet/iotex.json [hunk 2]
```diff
@@ -6,87 +6,36 @@
...
   "log_reconciliations": false,
   "ignore_reconciliation_error": false,
   "exempt_accounts": "",
-  "bootstrap_balances": "mainnet/bootstrap_balances.json",
-  "historical_balance_disabled": true,
-  "interesting_accounts": "mainnet/interesting_accounts.json",
+  "bootstrap_balances": "",
+  "interesting_accounts": "",
```
Comment (dustinxie) [inline rosetta-cli-config/mainnet/iotex.json:30]: why these are removed?

File: rosetta-cli-config/testing/bootstrap_balances.json [hunk 3]
```diff
@@ -49,4 +49,4 @@
     },
     "value": "80000000000000000000000000000000000"
   }
-]
+]
```
Comment (dustinxie) [inline rosetta-cli-config/testing/bootstrap_balances.json:52]: revert this

File: rosetta-cli-config/testnet/iotex.json [hunk 4]
```diff
@@ -6,87 +6,36 @@
...
+ "memory_limit_disabled": false,
+ "construction": null,
  "data": {
   "active_reconciliation_concurrency": 16,
   "inactive_reconciliation_concurrency": 4,
   "inactive_reconciliation_frequency": 250,
   "log_blocks": false,
   "log_transactions": false,
-  "log_balance_changes": true,
+  "log_balance_changes": false,
```
Comment (dustinxie) [inline rosetta-cli-config/testnet/iotex.json:25]: same

File: rosetta-cli-config/testnet/iotex.json [hunk 5]
```diff
@@ -6,87 +6,36 @@
...
   "log_reconciliations": false,
   "ignore_reconciliation_error": false,
   "exempt_accounts": "",
-  "bootstrap_balances": "testnet/bootstrap_balances.json",
-  "historical_balance_disabled": true,
-  "interesting_accounts": "testnet/interesting_accounts.json",
+  "bootstrap_balances": "",
+  "interesting_accounts": "",
```
Comment (dustinxie) [inline rosetta-cli-config/testnet/iotex.json:30]: same

File: services/account_test.go [hunk 6]
```diff
@@ -47,20 +47,13 @@ func TestAccountAPIService_AccountBalance(t *testing.T) {
 				Decimals: 18,
 			},
 		}
-		coin = &types.Coin{
-			CoinIdentifier: &types.CoinIdentifier{
-				Identifier: "IOTX",
-			},
-			Amount: amount,
-		}
```
Comment (dustinxie) [inline services/account_test.go:55]: why remove "IOTX", this is the native token right?

Comment (millken) [reply services/account_test.go:55]: &types.AccountBalanceResponse{} has removed `Coins`

File: rosetta-cli-config/testing/iotex-testing.json [hunk 7]
```diff
@@ -1,92 +1,93 @@
...
-  "change_scenario": null,
-  "clear_broadcasts": false,
-  "broadcast_behind_tip": false,
-  "block_broadcast_limit": 5,
-  "rebroadcast_all": false,
-  "new_account_probability": 0.5,
-  "max_addresses": 200
- }
-}
+
```
Comment (dustinxie) [inline rosetta-cli-config/testing/iotex-testing.json:93]: remove 2 empty lines

File: services/common.go [hunk 8]
```diff
@@ -56,3 +58,14 @@ func IsSupportedConstructionType(typ string) bool {
 	}
 	return false
 }
+
+func ConvertToIotexAddress(addr string) string {
```
Comment (dustinxie) [inline services/common.go:62]: why need add this func, to handle 0x format address?

File: tests/inject/inject_test.go [hunk 9]
```diff
@@ -255,39 +259,6 @@ func TestInjectTransferUseExecution(t *testing.T) {
...
-	fmt.Println("TestGetImplicitLog")
-	require := require.New(t)
-	conn, err := grpc.Dial(endpoint, grpc.WithInsecure())
-	require.NoError(err)
-	defer conn.Close()
-	acc, err := account.HexStringToAccount(privateKey)
-	require.NoError(err)
-	c := iotex.NewAuthedClient(iotexapi.NewAPIServiceClient(conn), acc)
-	for i := uint64(1); i < 120; i++ {
-		ret, err := c.API().GetImplicitTransferLogByBlockHeight(context.Background(),
```
Comment (dustinxie) [inline tests/inject/inject_test.go:270]: rename the test to TestGetTransactionLog, call
`c.API().GetTransactionLogByBlockHeight()`

File: services/common.go (hunk 8 above)
Comment (millken) [reply services/common.go:62]: yes

File: services/common.go [hunk 10]
```diff
@@ -56,3 +58,14 @@ func IsSupportedConstructionType(typ string) bool {
...
+	addr = strings.TrimSpace(addr)
+	if len(addr) == 41 && addr[:2] == "io" {
```
Comment (dustinxie) [inline services/common.go:64]: remove the if{} part, since it is normal ioxxxx address

File: tests/inject/go.mod [hunk 11]
```diff
@@ -1,16 +1,153 @@
...
-	github.com/iotexproject/go-pkgs v0.1.2-0.20200523040337-5f1d9ddaa8ee
-	github.com/iotexproject/iotex-address v0.2.1
-	github.com/iotexproject/iotex-antenna-go/v2 v2.3.2
-	github.com/iotexproject/iotex-core v0.8.1-0.20200713031334-9be4cb0f24ed
-	github.com/iotexproject/iotex-proto v0.3.1-0.20200708013036-e22eb581888c
-	github.com/stretchr/testify v1.4.0
-	google.golang.org/grpc v1.21.0
+	github.com/ethereum/go-ethereum v1.10.11
+	github.com/iotexproject/go-pkgs v0.1.6
+	github.com/iotexproject/iotex-address v0.2.5
```
Comment (dustinxie) [inline tests/inject/go.mod:8]: use latest 0.2.6

File: go.mod [hunk 12]
```diff
@@ -1,21 +1,159 @@
...
-	github.com/iotexproject/iotex-address v0.2.4
-	github.com/iotexproject/iotex-antenna-go/v2 v2.5.1-0.20210506215850-16f5d8cabc0a
-	github.com/iotexproject/iotex-core v1.1.4
-	github.com/iotexproject/iotex-proto v0.5.0
+	github.com/btcsuite/btcd v0.22.0-beta
+	github.com/coinbase/rosetta-sdk-go v0.7.0
+	github.com/golang/mock v1.6.0
+	github.com/golang/protobuf v1.5.2
+	github.com/iotexproject/go-pkgs v0.1.6
+	github.com/iotexproject/iotex-address v0.2.5
```
Comment (dustinxie) [inline go.mod:11]: use latest 0.2.6

File: go.mod [hunk 13]
```diff
@@ -1,21 +1,159 @@
...
-	github.com/iotexproject/go-pkgs v0.1.5-0.20210105202208-2dc9b27250a6
-	github.com/iotexproject/iotex-address v0.2.4
-	github.com/iotexproject/iotex-antenna-go/v2 v2.5.1-0.20210506215850-16f5d8cabc0a
-	github.com/iotexproject/iotex-core v1.1.4
-	github.com/iotexproject/iotex-proto v0.5.0
+	github.com/btcsuite/btcd v0.22.0-beta
+	github.com/coinbase/rosetta-sdk-go v0.7.0
+	github.com/golang/mock v1.6.0
+	github.com/golang/protobuf v1.5.2
+	github.com/iotexproject/go-pkgs v0.1.6
```
Comment (dustinxie) [inline go.mod:10]: use latest 0.1.10

File: go.mod [hunk 14]
```diff
@@ -1,21 +1,159 @@
...
+	github.com/iotexproject/iotex-address v0.2.5
+	github.com/iotexproject/iotex-antenna-go/v2 v2.5.1
+	github.com/iotexproject/iotex-core v1.6.0
+	github.com/iotexproject/iotex-proto v0.5.2
```
Comment (dustinxie) [inline go.mod:14]: use latest 0.5.4

File: services/common.go [hunk 15]
```diff
@@ -61,9 +61,7 @@ func IsSupportedConstructionType(typ string) bool {
 
 func ConvertToIotexAddress(addr string) string {
 	addr = strings.TrimSpace(addr)
-	if len(addr) == 41 && addr[:2] == "io" {
-		return addr
-	} else if len(addr) > 2 && (addr[:2] == "0x" || addr[:2] == "0X") {
+	if len(addr) > 2 && (addr[:2] == "0x" || addr[:2] == "0X") {
 		add, _ := address.FromHex(addr)
```
Comment (dustinxie) [inline services/common.go:65]: FromHex() may return error, need to check and return error here

**Review by dustinxie — APPROVED** [review]

**Pull request merged**